kube-derive/src/custom_resource.rs

```diff
@@ -261,6 +269,9 @@ impl CustomDerive for CustomResource {
                 derive_paths.push(syn::parse_str(d)?);
             }
         }
+        if apiextensions == "v1" {
+            derive_paths.push(syn::parse_str("::schemars::JsonSchema")?);
+        }
```

Comment:
`v1beta1` still allows schemas. Regardless, we should probably pull out these checks into a consistent `let enable_schemars` or something.

Maybe we should reverse course and only try to derive `JsonSchema` for the whole object if the spec derives it?

Reply:
Added the following for now.
```rust
// Schema generation is always enabled for v1 because it's mandatory.
// TODO Enable schema generation for v1beta1 if the spec derives `JsonSchema`.
let schema_gen_enabled = apiextensions == "v1";
```

Reply:
> Maybe we should reverse course and only try to derive JsonSchema for the whole object if the spec derives it?

Did you mean to do the same for `v1`? That can be nice for users not needing the schema generated. For example, if they need schemas with validations which `schemars` doesn't support at the moment. I think it's possible to generate using `#[schemars(with = "func")]` ([custom serialization](https://graham.cool/schemars/examples/7-custom_serialization/)), but it's not declarative.

The downside is that it can be misleading for `v1` users because `crd()` returns something they can't use directly.